flypipe/node_graph.py

```diff
@@ -107,22 +123,43 @@ def _build_graph(
                 graph.add_edge(dependency.key, current_transformation.key)
         graph = self._compute_requested_columns(graph)
         graph = self._expand_node_functions(graph)
-
         graph = self._compute_edge_selected_columns(graph)
+        graph = self._create_cache_contexts(graph, cache_context)
 
         for node_key in graph.nodes:
             transformation = graph.nodes[node_key]["transformation"]
             # TODO- move this to pandas_on_spark_node once we figure out how to get context to work
             # TODO- create a copy of the node, as in databricks it keeps the objects with type changed until the state
             # is cleared
             if (
-                pandas_on_spark_use_pandas
-                and transformation.dataframe_type == DataFrameType.PANDAS_ON_SPARK
+                    pandas_on_spark_use_pandas
+                    and transformation.dataframe_type == DataFrameType.PANDAS_ON_SPARK
             ):
                 transformation.type = "pandas"
 
         return graph
 
+    def _create_cache_contexts(self, graph, cache_context):
+        if cache_context:
+            for node_key in graph.nodes:
+                node = graph.nodes[node_key]
+                # should we overwrite the node cache?
+                # or should we create node['transformation'].cache_context = cache_context.create(node)?
+                node["transformation"].cache = cache_context.create(

```

Comment:
decided to overwrite node.cache (of type Cache) with an object of type CacheContext. it makes things easier  to deal with the same attribute cache inside node.py and also when calculating the run_status. otherwise we would have to deal with 2 attributes cache and cache_context that do the same thing, only difference is that cache_context holds spark